tests/nfs/nfs_ha_readdir_operation.py

```diff
@@ -0,0 +1,150 @@
+    file_count = int(config.get("file_count", "100"))
+    operations = config.get("operations")
+    Thread_operations = []
+    ha = bool(config.get("ha", False))
+    vip = config.get("vip", None)
+
+    # If the setup doesn't have required number of clients, exit.
+    if no_clients > len(clients):
+        raise ConfigError("The test requires more clients than available")
+
+    clients = clients[:no_clients]  # Select only the required number of clients
+    nfs_node = nfs_nodes[0]
+    fs_name = "cephfs"
+    nfs_name = "cephfs-nfs"
+    nfs_export = "/export"
+    nfs_mount = "/mnt/nfs"
+    fs = "cephfs"
+    nfs_server_name = nfs_node.hostname
+
+    try:
+        # Setup nfs cluster
+        setup_nfs_cluster(
+            clients,
+            nfs_server_name,
+            port,
+            version,
+            nfs_name,
+            nfs_mount,
+            fs_name,
+            nfs_export,
+            fs,
+            ha,
+            vip,
+        )
+
+        # Create oprtaions on each client
+        for client, operation in operations.items():
+            if operation == "create_files":
+                Thread_operations.append(
+                    Thread(
+                        target=create_files,
+                        args=(clients[int(client[-2:]) - 1], nfs_mount, file_count),
+                    )
+                )
+            elif operation == "remove_files":
+                Thread_operations.append(
+                    Thread(
+                        target=remove_files,
+                        args=(
+                            clients[int(client[-2:]) - 1],
+                            nfs_mount,
+                            file_count,
+                        ),
+                    )
+                )
+            elif operation == "perfrom_linux_untar":
+                Thread_operations.append(
+                    Thread(
+                        target=perfrom_linux_untar,
+                        args=(clients[int(client[-2:]) - 1], nfs_mount),
+                    )
+                )
+            elif operation == "perform_lookups":
+                Thread_operations.append(
+                    Thread(
+                        target=perform_lookups,
+                        args=(clients[int(client[-2:]) - 1], nfs_mount, file_count),
+                    )
+                )
+            elif operation == "perfrom_du_sh":
+                Thread_operations.append(
+                    Thread(
+                        target=perfrom_du_sh,
+                        args=(clients[int(client[-2:]) - 1], nfs_mount),
+                    )
+                )
+
+        # start opertaion on each client
+        for Thread_operation in Thread_operations:
+            Thread_operation.start()
+            sleep(3)
+
+        # Wait to complete operations
+        for Thread_operation in Thread_operations:
+            Thread_operation.join()
+
+        sleep(30)
+        # Reboot all clients
```

Comment:
Need to reboot server node having VIP assigned

Reply:
Updated